smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/commit/KafkaLatestCommit.java

```diff
@@ -18,13 +23,33 @@
 public class KafkaLatestCommit implements KafkaCommitHandler {
 
     private final KafkaConsumer<?, ?> consumer;
+    private final Map<TopicPartition, Long> offsets = new HashMap<>();
 
-    public KafkaLatestCommit(KafkaConsumer<?, ?> consumer) {
-        this.consumer = consumer;
+    public KafkaLatestCommit(io.vertx.mutiny.kafka.client.consumer.KafkaConsumer<?, ?> consumer) {
+        this.consumer = (KafkaConsumer<?, ?>) consumer.getDelegate();
     }
 
     @Override
     public <K, V> CompletionStage<Void> handle(IncomingKafkaRecord<K, V> record) {
-        return consumer.commit().subscribeAsCompletionStage();
+        CompletableFuture<Void> future = new CompletableFuture<>();
+        Map<TopicPartition, OffsetAndMetadata> map = new HashMap<>();
+        TopicPartition key = new TopicPartition(record.getTopic(), record.getPartition());
+        Long last = offsets.get(key);
+        // Verify that the latest committed offset before this one.
+        if (last == null || last < record.getOffset() + 1) {
+            offsets.put(key, record.getOffset() + 1);
+            map.put(key, new OffsetAndMetadata(record.getOffset() + 1, null));
+
+            consumer.commit(map, ar -> {
+                if (ar.failed()) {
+                    future.completeExceptionally(ar.cause());
+                } else {
+                    future.complete(null);
+                }
+            });
```

Comment:
This has always bothered me about this strategy. Committing on every message causes High load on the cluster. Maybe we should add a warning to the documentation.

Also, what' the implication of completing exceptionally here?

Reply:
it's actually better than the previous one that was committing on every message, every subscribed topic/partition. At least this one tracks the message more carefully. I've updated the doc.

Before switching to the throttled by default (which is the objective), we must do a lot more tests. Typically, it does not commit on application shutdown. 

About the completeExceptionally, if the commit fails, the failure is propagated downstream and mark the application as failed.

Reply:
Fixed.